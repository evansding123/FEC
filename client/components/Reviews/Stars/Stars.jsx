import React from 'react';
import PropTypes from 'prop-types';
import checkStar from '../func/checkStar/checkStar';

const Stars = (props) => {
  const {
    rate,
    classNameForSize,
    starsClicked,
  } = props;

  return (
    <div
      data-testid="stars"
      className="ratingHead"
    >
      {[...Array(5)].map((i, index) => (
        // eslint-disable-next-line react/no-array-index-key
        <div key={index}>
          <button
            data-testid={`button${index}`}
            type="button"
            id={index + 1}
            className="starButton"
            onClick={starsClicked}
          >
            .
          </button>
          <div className={classNameForSize}>
            <div
              className={`${classNameForSize} ${checkStar(index, rate)}`}
            />
          </div>
        </div>
      ))}
    </div>
  );
};

Stars.propTypes = {
  rate: PropTypes.number,
  classNameForSize: PropTypes.string,
  starsClicked: PropTypes.func,
};

Stars.defaultProps = {
  rate: 1,
  classNameForSize: '',
  starsClicked: () => 1,
};

export default Stars;
