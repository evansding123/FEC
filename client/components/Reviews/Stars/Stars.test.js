/**
 * @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
  screen,
} from '@testing-library/react';
import Stars from './Stars';

afterEach(cleanup);

it('Stars will trigger when clicked', () => {
  const testFunc = jest.fn();
  render(<Stars starsClicked={testFunc} />);
  fireEvent.click(screen.getByTestId('button0'));
  fireEvent.click(screen.getByTestId('button1'));
  fireEvent.click(screen.getByTestId('button2'));
  fireEvent.click(screen.getByTestId('button3'));
  fireEvent.click(screen.getByTestId('button4'));
  expect(testFunc).toHaveBeenCalledTimes(5);
});

it('should have default starsClicked()', () => {
  expect(Stars.defaultProps.starsClicked()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <Stars
      rate={1}
      classNameForSize="hi"
      starsClicked={jest.fn}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
