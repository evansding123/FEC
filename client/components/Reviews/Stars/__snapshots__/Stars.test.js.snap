// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="ratingHead"
  data-testid="stars"
>
  <div>
    <button
      className="starButton"
      data-testid="button0"
      id={1}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <div
      className="hi"
    >
      <div
        className="hi fullStar"
      />
    </div>
  </div>
  <div>
    <button
      className="starButton"
      data-testid="button1"
      id={2}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <div
      className="hi"
    >
      <div
        className="hi emptyStar"
      />
    </div>
  </div>
  <div>
    <button
      className="starButton"
      data-testid="button2"
      id={3}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <div
      className="hi"
    >
      <div
        className="hi emptyStar"
      />
    </div>
  </div>
  <div>
    <button
      className="starButton"
      data-testid="button3"
      id={4}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <div
      className="hi"
    >
      <div
        className="hi emptyStar"
      />
    </div>
  </div>
  <div>
    <button
      className="starButton"
      data-testid="button4"
      id={5}
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <div
      className="hi"
    >
      <div
        className="hi emptyStar"
      />
    </div>
  </div>
</div>
`;
